landscape/client/manager/snapmanager.py

```diff
@@ -292,6 +292,27 @@ def run(self):
             self._send_snap_update,
         )
 
+    def _get_serial(self):
+        """
+        Returns device's serial number by parsing the serial assertion.
+
+        If the serial assertion cannot be retrieved, returns "no-serial".
+        """
+        serial = "no-serial"
+
+        try:
+            serial_assert = snap_http.get_assertions("serial").result
+        except SnapdHttpException as e:
+            logging.error(f"Unable to get serial assertion: {e}")
+            return serial
+
+        for line in serial_assert.splitlines():
+            if line.startswith("serial:"):
+                serial = line.split("serial:", 1)[1].strip()
```

Comment:
In the unlikely event that `line` is just `"serial:"`, then this function will return `""` and the snap ID will be `-{snap_name}`.

Are we okay with that. How likely is it? Should we guard against it?

Reply:
A quick fix is to end with:
```python
return serial or "no-serial"
```